### ISSUE TITLE
Download response as a file with Content-Disposition: attachment not available in RESTer v. 4.11.0

### ISSUE THREAD

**Issue by parvizfarnia** [frigus02/RESTer#962]
Hello

In the [download page of RESTer on Mozilla](https://addons.mozilla.org/en-US/firefox/addon/rester/), at the bottom of the page, there is a release note:

> Release notes for 4.11.0
> Added
> 
> - RESTer will now download response bodies as files, if the response has a `Content-Disposition: attachment` header. 

I have installed the 4.11.0 version, yet I don't see this option available.

Is that normal?

Thanks in advance.

**Comment by frigus02** [comment]
Sorry for the slow response. There is no visible option for it in the UI. RESTer should open a file save dialog if the response contains the appropriate header, e.g. ` Content-Disposition: attachment; filename=hello.txt `.

If this doesn't work for you, it sounds like a bug. Can you share the response headers where you expected RESTer to download the response as a file?

**Comment by parvizfarnia** [comment]
Thank you for your help.

Actually, I'm not even sure that I'm sending the query correctly. Because looking the value you specified, I never put any filename.

Here is the context. I use RESTer to query a REST API. Here is what I have in my header inside RESTer screen when I send my query

```
Method: GET
Authorization: Bearer <And then I write my token>
Accept: application/json
```

Now sometimes, I need to get the result in CSV format instead of JSON. In that case, I write `Accept: text/csv`. But in this way, I don't get the result in a file and I have to simply copy/paste the result from RESTer screen into Excel/Calc. But recently I noticed that apparently there is `Content-Disposition` functionality. I said, well that would be quite more appropriate to use instead of doing each time a copy/paste.

As you suggested I tried to add the following to my query header:

`Content-Disposition: attachment; filename=hello.txt`

But nothing happend in terms of downloading. The query succeeded as usual (HTTP code 200) but that's all. Here is the response header

```
Cache-Control: no-cache
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: POST,GET
Content-Encoding: gzip
Vary: Accept-Encoding
Expires: Thu, 01 Jan 1970 00:00:00 GMT
X-Total-Count: 26329955
Access-Control-Allow-Headers: authorization,Access-Control-Allow-Origin,Content-Type,SOAPAction
Link: <I erased the content here as it is confidential>
Content-Type: text/csv;charset=utf-8
Date: Tue, 13 Jun 2023 07:57:21 GMT
Transfer-Encoding: chunked
Connection: keep-alive
Set-Cookie: pdapimgateway=<I also erased the content here>...; expires=Tue, 13-Jun-2023 08:17:21 GMT; path=/; Httponly; Secure
Strict-Transport-Security: max-age=100000; includeSubDomains
```

**Comment by parvizfarnia** [comment]
Yes. Besides, unless I have not understood how RESTer functions, it seems to me that  currently a user has only the possibility of setting request headers in RESTer's UI. Am I right?

**Comment by frigus02** [comment]
Yes, RESTer can only set request headers. `Content-Disposition` is a response header only the responding server can set it.

I'll take this as a feature request to download any kind of response as a file.
